Mathlib/Data/Set/Finite/Basic.lean

```diff
@@ -156,7 +156,7 @@ protected alias ⟨_, toFinset_strictMono⟩ := Finite.toFinset_ssubset_toFinset
 
 @[simp high]
 protected theorem toFinset_setOf [Fintype α] (p : α → Prop) [DecidablePred p]
-    (h : { x | p x }.Finite) : h.toFinset = Finset.univ.filter p := by simp
+    (h : { x | p x }.Finite) : h.toFinset = ({x | p x} : Finset α) := by simp
```

Comment:
@YaelDillies OTOH, this type annotation is required, otherwise Lean decides to cast LHS to `Set α` instead.